### ISSUE TITLE
Quickstart validation hitting rate limiting

### ISSUE THREAD

**Issue by aswanson-nr** [newrelic/newrelic-quickstarts#783]
## Description
The new Nerdgraph validation for quickstarts is hitting API rate limits for large pull requests. 

## Steps to Reproduce
1. Have a PR with lots of changed quickstarts, such as #779 
2. Notice that the validation against nerdgraph starts to throw errors, [here](https://github.com/newrelic/newrelic-quickstarts/runs/5038388168?check_suite_focus=true)

## Expected Behavior
- [ ] The validation workflow is able to handle any number of quickstarts in a pull request.

**Comment by aswanson-nr** [comment]
Comment from @nr-kkenney:
> i heard mention of throttling / handling errors, and thought this might be another opportunity to try out [cockatiel](https://github.com/connor4312/cockatiel) (or a similar solution).
> 
> I had an example usage PR from before: https://github.com/newrelic/docs-website/pull/3219/files.
> 
> tl;dr -- In the simplest case, we can retry requests & implement exponential backoff with jitter for each request very easily. If it's the case that we simply make too many requests (and we shouldn't make more), there are also other built in patterns that we can utilize like circuit breaker.
> 
> Not sure if there is a strict rate limiting / throttling functionality, but I assume we still want some retry functionality in place.

**Comment by jpvajda** [comment]
@aswanson-nr would you be able to put an estimate on this 🐛 ?
